### ISSUE TITLE
User Ranking

### ISSUE THREAD

**Issue by NathanaelA** [NativeScript/plugins#5]
I think we should allow some user ranking without logging in; however I think we might need to decide a criteria.

-  If the unlogged in user starts marking a ton of plugins on the same day, we ask them to log in on say the forth one.    

- If the unlogged in user only marks one or two plugins a day, we are fine -- this is probably legit.  I know I tend to download a couple plugins to play with and then decide which one I like.

- We need to keep track of every ranking by every person/ip address, the idea is that if someone attempts to do market manipulation we can just disregard everything they did.  (i.e. shadow/perm-ban the account?)

- If the user has voted on an plugin, show their ranking in a difference color below/besides the actual current ranking, and let them change the ranking at any time.

**Comment by georgeedwards** [comment]
Not sure about this, Why not get them to sign in? We would have to implement the login functionality anyway, if we use GitHub OAuth2, then everyone will already have an account, then we can protect all APIs fro DDoS etc. + easily link the actions of all users without worrying about ip etc.

**Comment by NathanaelA** [comment]
I understand your point; and I've been considering this.   That is why I brought this up. 

Actually we are discussing end users (not plugin authors) ; a lot of the end users won't have a github account.  Which is why I mentioned Google/Facebook/Twitter as the login auth integration.

But even if you have someone log in you still have to track clicks per user to eliminate ranking issues.    So that isn't anything extra.  The bigger thing is that the odds of people ranking without logging in are a lot greater than making someone login.  The less work we make them do; the better the odd's we will get more rankings per plugin...   ;-)    However the trade off is that you open up for slightly more plugin ranking manipulation...   That is the question, are we willing to take this trade off to get what I believe is a LOT more rankings...

**Comment by georgeedwards** [comment]
Fair point, I personally would air on the side of caution, because if users notice even one manipulation before we fix it, then they would likely stop trusting all rankings, invalidating a whole load of them. I agree we would likely get a lot more rankings. Logging-in definitely protects against a user using multiple ip addresses.

Would be good to hear anyone else's view.

**Comment by bradmartin** [comment]
Tough subject :) I'd say I'm leaning toward auth with social. Couple clicks
and then can rank so less likely someone to go around spamming. Plus if
they can rank/rate without login you have some people possibly rating
negatively for no good reason. Of course there are a hundred reasons for
and against so then again implementation time to add the social auth is a
factor. I don't imagine many people will spam the system negatively but
there's always some. If there are negative rankings it would be nice to get
feedback for low/negative ratings so others can see why something was rated
negatively. Maybe a required message of xx characters and then of course we
as a community can audit the trolls/spammers ratings and remove from the
system. I'm rambling so have a good one everybody. Really excited for this
and will help out if anything's needed.

**Comment by NathanaelA** [comment]
I really like the idea of requiring a message for a low ranking <= 2 on a 1-5 scale...  That would be very helpful to the plugin author....
